src/main/kotlin/org/jetbrains/plugins/bsp/extension/points/PythonSdkGetterExtension.kt

```diff
@@ -0,0 +1,63 @@
+package org.jetbrains.plugins.bsp.extension.points
+
+import com.intellij.openapi.extensions.ExtensionPointName
+import com.intellij.openapi.projectRoots.ProjectJdkTable
+import com.intellij.openapi.projectRoots.Sdk
+import com.intellij.openapi.projectRoots.impl.SdkConfigurationUtil
+import com.intellij.platform.backend.workspace.virtualFile
+import com.intellij.platform.workspace.storage.impl.url.toVirtualFileUrl
+import com.intellij.platform.workspace.storage.url.VirtualFileUrlManager
+import com.jetbrains.python.sdk.PythonSdkAdditionalData
+import com.jetbrains.python.sdk.PythonSdkType
+import org.jetbrains.plugins.bsp.server.tasks.PythonSdk
+import java.net.URI
+import kotlin.io.path.toPath
+
+public interface PythonSdkGetterExtension {
+
+  public companion object {
```

Comment:
it's a performance thing, now it's recommended to put such things as top level declarations - so we should move it out of the companion object just to the file, + in this scenario we prob dont need the `extensions` - it should be enough to get just nullable impl instead of a whole list